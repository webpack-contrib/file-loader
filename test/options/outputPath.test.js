/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('outputPath', () => {
    test('{String}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath: 'output_path/',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{String} without trailing slash', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath: 'output_path',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{String} with `options.name`', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            name: '[path][name].[ext]',
            outputPath: 'output_path/',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{Function}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath(url) {
              return `output_path_func/${url}`;
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{Function} with `options.name`', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            name: '[name].[ext]',
            outputPath(url) {
              return `output_path_func/${url}`;
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{String} with `publicPath` {String}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath: 'output_path/',
            publicPath: 'public_path/',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{String} with `publicPath` {String} without trailing slash', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath: 'output_path',
            publicPath: 'public_path',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{Function} with `publicPath` {String}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            publicPath: 'public_path/',
            outputPath(url) {
              return `output_path_func/${url}`;
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{String} with `publicPath` {Function}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath: 'output_path/',
            publicPath(url) {
              return `public_path_func/${url}`;
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });

    test('{Function} with `publicPath` {Function}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath(url) {
              return `output_path_func/${url}`;
            },
            publicPath(url) {
              return `public_path_func/${url}`;
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { assets, source } = stats.toJson().modules[1];

      expect({ assets, source }).toMatchSnapshot();
    });
  });
});
