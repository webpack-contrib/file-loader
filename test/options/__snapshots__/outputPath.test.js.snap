// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options outputPath {Function} 1`] = `
Object {
  "assets": Array [
    "output_path_func/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"output_path_func/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options outputPath {Function} with \`options.name\` 1`] = `
Object {
  "assets": Array [
    "output_path_func/file.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"output_path_func/file.png\\";",
}
`;

exports[`Options outputPath {Function} with \`publicPath\` {Function} 1`] = `
Object {
  "assets": Array [
    "output_path_func/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path_func/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options outputPath {Function} with \`publicPath\` {String} 1`] = `
Object {
  "assets": Array [
    "output_path_func/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options outputPath {String} 1`] = `
Object {
  "assets": Array [
    "output_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"output_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options outputPath {String} with \`options.name\` 1`] = `
Object {
  "assets": Array [
    "output_path/file.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"output_path/file.png\\";",
}
`;

exports[`Options outputPath {String} with \`publicPath\` {Function} 1`] = `
Object {
  "assets": Array [
    "output_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path_func/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options outputPath {String} with \`publicPath\` {String} 1`] = `
Object {
  "assets": Array [
    "output_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options outputPath {String} with \`publicPath\` {String} without trailing slash 1`] = `
Object {
  "assets": Array [
    "output_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = \\"public_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`Options outputPath {String} without trailing slash 1`] = `
Object {
  "assets": Array [
    "output_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"output_path/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;
